annubes/task.py

```diff
@@ -470,8 +470,12 @@ def _build_trials_inputs(self) -> NDArray[np.float64]:
                 (len(self._time[n]), self._n_inputs),
                 dtype=np.float32,
             )
-            for idx, _ in enumerate(self._modalities):
-                value = self._rng.choice(self.stim_intensities, 1) if self._modality_seq[n] != "X" else 0
+            for idx, mod in enumerate(self._modalities):
+                value = (
+                    self._rng.choice(self.stim_intensities, 1)
+                    if (self._modality_seq[n] != "X" and self._modality_seq[n] == mod)
+                    else 0
+                )
```

Comment:
Given that the condition is more complicated now, I would find one of these options more readable.

```suggestion
            for idx, mod in enumerate(self._modalities):
                if self._modality_seq[n] == "X" or self._modality_seq[n] != mod:
                    value = 0
                else:
                    value = self._rng.choice(self.stim_intensities, 1)
                )
```

```suggestion
            for idx, mod in enumerate(self._modalities):
                value = 0
                if self._modality_seq[n] != "X" and self._modality_seq[n] == mod:
                    value = self._rng.choice(self.stim_intensities, 1)
```

Reply:
OK, looking more closely at the function, if I read it correctly, the place where value is used is defaulted to 0 anyway by the lines above
```python
            x[n] = np.zeros(
                (len(self._time[n]), self._n_inputs),
                dtype=np.float32,
            )
```
I guess that means we can drop the `value` variable and immediately re-assign `x[n][self._phases[n]["input"], idx]` in line 478 if the conditions apply. 
In fact it is defaulted to a float 0 and here re-assigned to an int 0. Not sure whether this matters at all, but just noticed the minor inconsistency.

Reply:
Good point! I've changed it as you suggested